### PR TITLE
[StableDiffusionInpaintPipeline] accept tensors for init and mask image

### LINKED ISSUE #370: Lack of functionality in stable diffusion inpainting
The img2img can handle init_image as FloatTensor and PIL image with
```
if not isinstance(init_image, torch.FloatTensor):
   init_image = preprocess(init_image)
```
But inpainting cannot work like this with init_image and mask_image. It's only working with FloatTensor. Although the typing states that it can work with both. 
```
init_image: Union[torch.FloatTensor, PIL.Image.Image],
mask_image: Union[torch.FloatTensor, PIL.Image.Image],
```
It is desirable to have uniformity.
Also, preprocess function in img2img and preprocess_image function in inpainting are the same, they differ only in the name.

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR updates `StableDiffusionInpaintPipeline` to accept both `torch.FloatTensor` and `PIL.Image.Image` for `init_image` and `mask_image`.

Fixes #370

### CODE REVIEW

**Comment by Inkorak** [comment]
There are problem here, in my opinion. If we send a FloatTensor, then there will be no preprocessing and the mask variable will not be declared, and secondly, there will be no transfer to the device.
```
if not isinstance(mask_image, torch.FloatTensor):
   mask = preprocess_mask(mask_image).to(self.device)
mask = torch.cat([mask] * batch_size)
```
With this, there should be no problems:

```
if not isinstance(mask_image, torch.FloatTensor):
   mask_image = preprocess_mask(mask_image)
mask = torch.cat([mask_image.to(self.device)] * batch_size)
```

**Comment by patil-suraj** [comment]
good catch! Updating it now

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py [hunk 1]
```diff
@@ -202,7 +202,8 @@ def __call__(
         self.scheduler.set_timesteps(num_inference_steps, **extra_set_kwargs)
 
         # preprocess image
-        init_image = preprocess_image(init_image).to(self.device)
+        if not isinstance(init_image, torch.FloatTensor):
+            init_image = preprocess_image(init_image)
```
Comment (pcuenca) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:206]: Why not put the `.to(self.device)` here, for symmetry with the mask and perhaps a bit more clarity? Currently it's a bit "hidden" a couple of lines below inside the call to `self.vae.encode()`.

Comment (patil-suraj) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py:206]: good point! Updated this to make device placement more clear. It's now after the if block for both mask and init image.

**Review by pcuenca — APPROVED** [review]
LGTM!

Question: if the user provides an image and a mask as tensors, should we verify that the number of channels match? The image should have R,G,B while the mask is only L. Is this something we should check or is it too much?

On second thought, we could just update the documentation and not the code:

```
mask_image (`torch.FloatTensor` or `PIL.Image.Image`):
    `Image`, or tensor representing an image batch, to mask `init_image`. White pixels in the mask will be
    replaced by noise and therefore repainted, while black pixels will be preserved. If `mask_image` is a
    PIL image, it will be converted to a single channel (luminance) before use. If it's a tensor, it should
    contain one color channel (L) instead of 3, so the expected shape would be `(B, H, W, 1)`.
```

**Review by patrickvonplaten — APPROVED** [review]

**Comment by patil-suraj** [comment]
Good point @pcuenca ! The pipeline is experimental and will be soon updated so just updated the docs for now.
